src/components/smart_objects/test/ArraySchemaItem_test.cc

```diff
@@ -345,6 +349,92 @@ TEST(test_map_validate, test_ArraySchemaItemTest) {
   EXPECT_EQ(std::string("Out of array"), obj["array"][4].asString());
 }
 
+TEST(test_array_with_unknown_enum, test_ArraySchemaItemTest) {
+  using namespace ns_smart_device_link::ns_smart_objects;
+  SmartObject obj;
+  std::set<ExampleEnum::eType> enum_values;
+  enum_values.insert(ExampleEnum::Value0);
+  enum_values.insert(ExampleEnum::Value1);
+  enum_values.insert(ExampleEnum::Value2);
+
+  ISchemaItemPtr item = CArraySchemaItem::create(
+      TEnumSchemaItem<ExampleEnum::eType>::create(enum_values),
+      TSchemaItemParameter<size_t>(1));
+
+  obj[0] = "Value0";
+  obj[1] = "Value10";
+  obj[2] = "Value1";
+
+  item->applySchema(obj, true);
+
+  rpc::ValidationReport report("RPC");
+  EXPECT_FALSE(
+      item->filterInvalidEnums(obj, utils::SemanticVersion(), &report));
+  EXPECT_NE(std::string(""), rpc::PrettyFormat(report));
+
+  EXPECT_EQ((size_t)2, obj.length());
+  EXPECT_EQ(0, obj[0].asInt());
+  EXPECT_EQ(1, obj[1].asInt());
+
+  report = rpc::ValidationReport("RPC");
+  EXPECT_EQ(errors::OK, item->validate(obj, &report));
+  EXPECT_EQ(std::string(""), rpc::PrettyFormat(report));
+}
+
+TEST(test_array_with_unknown_enums, test_ArraySchemaItemTest) {
+  using namespace ns_smart_device_link::ns_smart_objects;
+  SmartObject obj;
+  std::set<ExampleEnum::eType> enum_values;
+  enum_values.insert(ExampleEnum::Value0);
+  enum_values.insert(ExampleEnum::Value1);
+  enum_values.insert(ExampleEnum::Value2);
+
+  ISchemaItemPtr item = CArraySchemaItem::create(
+      TEnumSchemaItem<ExampleEnum::eType>::create(enum_values),
+      TSchemaItemParameter<size_t>(1));
+
+  obj[0] = "Value10";
+  obj[1] = "Value11";
+  obj[2] = "Value12";
+
+  item->applySchema(obj, true);
+
+  rpc::ValidationReport report("RPC");
+  EXPECT_TRUE(item->filterInvalidEnums(obj, utils::SemanticVersion(), &report));
+  EXPECT_NE(std::string(""), rpc::PrettyFormat(report));
+
+  EXPECT_EQ((size_t)0, obj.length());
+
+  report = rpc::ValidationReport("RPC");
+  EXPECT_EQ(errors::OUT_OF_RANGE, item->validate(obj, &report));
+  EXPECT_NE(std::string(""), rpc::PrettyFormat(report));
```

Comment:
Why do we expect empty report?

Reply:
We expect a non-empty report (`EXPECT_NE`). I don't think we should enforce a strict format, just verify that some warning was generated